pkgs/development/haskell-modules/configuration-hackage2nix.yaml

```diff
@@ -2537,6 +2537,7 @@ default-package-overrides:
 
 extra-packages:
   - aeson < 0.8                         # newer versions don't work with GHC 7.6.x or earlier
+  - Agda == 2.6.1                       # allows agdaPackages to depend on a fixed version
```

Comment:
Cause still is not understandable, it is not reached in the explanation.

Contributors still do not know the cause. Maybe there was some particular quirk issue and because of it `agda` was fixed.

As you said in the Report text: "_ so that the current package set can be fixed to this version and won't break when another version comes out._, pretty good explanation.

Reply:
I was trying not to make it too long but I guess this doesn't matter